docs/user/reconciliation.md

```diff
@@ -164,3 +164,10 @@ For subsequent updates, the builder blocks until:
 ### Job
 
 After pushing the merged Job into the cluster, the builder blocks until the Job succeeds.
+
+Since neither the use case requiring a Job's spec.selector to be explicitly set nor the handling thereof is well defined the code will panic if a user attempts to set spec.selector.
+
+Subsequent updates:
+
+* Job update handling is not well-defined and is therefore not recommended.
+* A Job's spec.selector is immutable so a Job's spec.selector will never be updated.
```

Comment:
I think this point is covered in your earlier `selector` sentence, so we can drop the reference here.

Reply:
Clarified a bit but left it here (see https://github.com/openshift/cluster-version-operator/pull/312#discussion_r375605029)

Reply:
> Clarified a bit but left it here...

6b58e7a wording looks good to me.